frontend/test/metabase/meta/Dashboard.unit.spec.js

```diff
@@ -399,6 +399,27 @@ describe("meta/Dashboard", () => {
             dimensions: {},
             fieldValues: () => [],
           },
+          [[
```

Comment:
This ends up getting coerced into a string via `toString`, so this becomes the key `'field,boolean,[object Object]'` on the `metadata.fields` map. I see that it is already a problem on `master`, so I won't block this bug fix, but I bet handling virtual field IDs on the `metadata` object better would make some of that mapping code much nicer. I'm guessing we'll need to add code to add these fields to a separate map or do something like set them at a special key like `fieldId + tableId` (tableId in this scenario looks like `"card_123"`).

Reply:
Yep, it should be something like `fieldId + tableId` because there is also another problem that these `field,boolean,[object Object]` strings are not unique. 

I checked how many tests will fail if I use `fieldId` there instead of object, and it was a lot: https://github.com/metabase/metabase/pull/18229